discovery-infra/junit_log_parser.py

```diff
@@ -83,6 +84,40 @@ def export_service_logs_to_junit_suites(source_dir: Path, report_dir: Path):
         f.write(xml_report)
 
 
+def get_event_test_case(event: dict):
+    event_type = event["type"]
+    if event_type.lower() in EXPORTED_EVENT_LEVELS:
+        log.info(f"Adding {event_type} event as test-case")
+        reason = event["reason"]
+        involved_object = event["involvedObject"]
+        obj = involved_object["kind"].lower() + "/" + involved_object["name"]
+        message = event["message"]
+        timestamp = event["firstTimestamp"]
+        test_case = TestCase(name=obj, classname="EVENT", category=reason, timestamp=timestamp)
+        test_case.failures.append(CaseFailure(message=message, output=message, type=event_type))
+        return test_case
+    return None
+
+
+def export_service_events_to_junit_suite(source_dir: Path, report_dir: Path, events_file_name="k8s_events.json"):
+    test_cases = list()
+    with open(source_dir.joinpath(events_file_name)) as f:
+        events_data = json.load(f)
```

Comment:
also here, why should we save all ``events_data`` in memory?

Reply:
It's cannot be done. Only with `JSON lines format` (we can't convert it without open the entire file)